### ISSUE TITLE
Non-determinism during viewer initialization.

### ISSUE THREAD

**Issue by yurydelendik** [mozilla/pdf.js#7797]
There are some initialization issues can be present in time between module loading and viewer start up (e.g. https://github.com/mozilla/pdf.js/issues/7448). We might still have similar issues. We need to remove/replace all {document|window}.addEventListener and Promise from the main initialization path, I found so far:

```
document.addEventListener`DOMContentLoaded @http://localhost:8888/web/viewer.js:218:1
Promise @http://localhost:8888/web/preferences.js:34:24
window.addEventListener`resize @http://localhost:8888/web/app.js:1803:1
window.addEventListener`hashchange @http://localhost:8888/web/app.js:1828:1
window.addEventListener`change @http://localhost:8888/web/app.js:1849:3
window.addEventListener`localized @http://localhost:8888/web/app.js:1887:1
window.addEventListener`wheel @http://localhost:8888/web/app.js:2053:1
window.addEventListener`click @http://localhost:8888/web/app.js:2055:1
window.addEventListener`keydown @http://localhost:8888/web/app.js:2067:1
window.addEventListener`beforeprint @http://localhost:8888/web/app.js:2327:1
window.addEventListener`afterprint @http://localhost:8888/web/app.js:2331:1
Promise animationStartedClosure@http://localhost:8888/web/app.js:2338:50
window.addEventListener`keydown @http://localhost:8888/web/pdf_print_service.js:285:3
window.addEventListener`beforeprint @http://localhost:8888/web/pdf_print_service.js:323:5
window.addEventListener`afterprint @http://localhost:8888/web/pdf_print_service.js:324:5
```

(I also wonder if we have something like `window.onload = ...` on the module initialization path)

The code above must be moved into some initialization routines.

To deal with one-time call addEventListener (e.g. DOMContentLoaded): if possible, we need to check state and if state was not triggered use addEventListener.

**Comment by yurydelendik** [comment]
Script used the generate report above:

```diff
diff --git a/web/viewer.js b/web/viewer.js
index ab620a2..565ada3 100644
--- a/web/viewer.js
+++ b/web/viewer.js
@@ -16,6 +16,33 @@
 
 'use strict';
 
+function __log(s) { console.log(s + ' ' + (new Error().stack.split('\n')[2])); }
+var __oldDocumentAddEventListener = document.addEventListener;
+document.addEventListener = function (t) {
+  __log('document.addEventListener`' + t);
+  return __oldDocumentAddEventListener.apply(this, arguments);
+};
+var __oldWindowAddEventListener = window.addEventListener;
+window.addEventListener = function (t) {
+  __log('window.addEventListener`' + t);
+  return __oldWindowAddEventListener.apply(this, arguments);
+};
+var __oldPromise = window.Promise;
+window.Promise = function(resolve, reject) {
+  __log('Promise');
+  return new __oldPromise(resolve, reject);
+};
+window.Promise.prototype = {
+  all: function () { __log('Promise.all'); __oldPromise.prototype.all.apply(__oldPromise, arguments); },
+  resolve: function () { __log('Promise.resolve'); __oldPromise.prototype.resolve.apply(__oldPromise, arguments); },
+  reject: function () { __log('Promise.reject'); __oldPromise.prototype.reject.apply(__oldPromise, arguments); },
+};
+function __reset() {
+  window.Promise = __oldPromise;
+  window.addEventListener = __oldWindowAddEventListener;
+  document.addEventListener = __oldDocumentAddEventListener;
+}
+
 var DEFAULT_URL = 'compressed.tracemonkey-pldi-09.pdf';
 
 if (typeof PDFJSDev !== 'undefined' && PDFJSDev.test('CHROME')) {
@@ -178,6 +205,7 @@ function webViewerLoad() {
       // issue 7448).
       require(['pdfjs-web/app', 'pdfjs-web/pdf_print_service'], function (web) {
         window.PDFViewerApplication = web.PDFViewerApplication;
+        __reset();
         web.PDFViewerApplication.run(config);
       });
     });
```

**Comment by timvandermeij** [comment]
Partial fix in #7886.
